src/lib/eloCalculator.js

```diff
@@ -49,3 +48,27 @@ export function calculateAllElo(txt) {
       {}
     );
 }
+
+
+export function calculateEloOverTime(txt) {
+  var currentElo = []
+  return txt.trim().split("\n").slice(1)
+    .reduce(
+      (elosAllTime, line) => {
+        const [white, black, outcome, year, month, day] = line.split(",");
+        const result = eloCalc(
+          currentElo[white] || 1000,
+          currentElo[black] || 1000,
+          outcome
+        );
+        currentElo[white] = result[0];
+        currentElo[black] = result[1];
+        elosAllTime[white] = elosAllTime[white] || [];
+        elosAllTime[black] = elosAllTime[black] || [];
+        elosAllTime[white].push([Date.UTC(year, month, day),result[0]]);
+        elosAllTime[black].push([Date.UTC(year, month, day),result[1]]);
```

Comment:
Have you already considered using ISO date?

`Date.Parse()` looks like it takes ISO date, which is more standardized than `yy,mm,dd`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse

Also important if two people play more than once that day.

Reply:
I saw a Highchart example that used Date.UTC so I just went with it

ISO dates look a lot more user friendly than requiring `yyyy,mm,dd`

With this, we only have daily granularity, so if someone plays twice in a day, they have 2 data points on one day as seen here https://deploy-preview-27--chesselo.netlify.com/

Reply:
> they have 2 data points on one day as seen here
That looks... funny. I'm guessing it can't be avoided unless we get more granular in the x-axis?

Also, I assume this is WIP and the line graph will be per ruleset?

Reply:
I'm thinking there's going to be some decent sized UI changes needed.

Showing a leaderboard graph + leaderboard table + elo over time graph for multiple rulesets at the same time isn't feasible imo

Reply:
Which ruleset does the line graph represent?

Reply:
withDate.txt (https://github.com/turtleman270/Chess-Leaderboards/blob/master/public/data/withDate.txt)

There should only be one ruleset format that's used for each ruleset. Renaming them to `5MinuteGames` or something similar would probably help me get less confused